### ISSUE TITLE
The `function-disallowed-list` rule does not report an error in the return expression of a scss function

### ISSUE THREAD

**Issue by sharikovvladislav** [stylelint/stylelint#8565]
### What minimal example or steps are needed to reproduce the bug?

```
@use "sass:math"
@function get-random-value() {
  @return math.random(100.5);
}
.a { margin-left: get-random-value(); }
```
^^ this does not work

However, the following works:
```
@use "sass:math"
@function get-random-value() {
  $v: math.random(100.5);
  @return $v;
}
.a { margin-left: get-random-value(); }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "scss/function-disallowed-list": [["random"]],
  }
}
```

### How did you run Stylelint?

```
stylelint app/**/*.scss --config .stylelintrc.json
```

### Which Stylelint-related dependencies are you using?

```json
"stylelint": "16.8.1",
"stylelint-config-standard": "36.0.1",
"stylelint-order": "6.0.4",
"stylelint-prettier": "5.0.3",
"stylelint-scss": "6.5.0",
```

### What did you expect to happen?

I expect that the `scss/function-disallowed-list` will be violated because of the config. 

### What actually happened?

The error was not reported when the code with a violation is the return expression of a function.

### Do you have a proposal to fix the bug?

Here is the reproduction. Add the following test to the `src/rules/function-disallowed-list/__tests__/index.js` spec file:
```
testRule({
  ruleName,
  config: ["random"],
  customSyntax: "postcss-scss",

  reject: [
    {
      code: `
        @use "sass:math"
        @function get-random-value() {
          @return math.random(100.5);
        }
        .a { margin-left: get-random-value(); }
      `,
      message: messages.rejected("random"),
      description: "Math library function, not allowed."
    }
  ],
});
```

**Comment by ybiquitous** [comment]
@sharikovvladislav Thank you for the report with the reproduction. I can also reproduce the bug with [this online demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAzArgOzAFwJYQwAIBzGbAWgCcBDDKCAW3IDdqAbNGACgEpDgAOkUJJKZNJSINq2ABYA6GnUZcAjAAZ18gKw8A3EKGFCAEmbxC0uYtr0GazTv1GRY7BKJmDGAL5CQADQgKLhsMABy1AxwiDAAHlEADmHyAM5gqamB4AQhxAgggsICIMloxLgYqSUWANolqdgAnmFslRTpmSUAugEuJWBojYwAyk0Y2NRxNYQliRCNGankndWB-SCUaGFrFkXGxg1LAPToWHgE5FC4qexsEADuMFDkbY0ztfWbtow93S5+XzZSAYPIAMQglCsBQAVqkCNlYIksoh9rNSgtsEsVksZiUACzyLQAThKfWKIEaLRgbQmeJAqgAbPIABzyVRkjZU1rtHFdBDo5naIklIQ+EA+IA).

Unfortunately, this problem seems to be caused by not a `stylelint` built-in rules but the `stylelint-scss` rule. I'm not 100% sure, but the cause looks like that `scss/function-disallowed-list` handles only declarations, not at-rules:

https://github.com/stylelint-scss/stylelint-scss/blob/ae0af5eb4001fe3869f8fcbf9760f4d62f432e3f/src/rules/function-disallowed-list/index.js#L33

`postcss-scss` considers the following code as an *at-rule*,

```scss
@return math.random(100.5);
```

while it considers the following code as a *declration*:

```scss
$v: math.random(100.5);
```

If you find my guess right, can you please open an issue on <https://github.com/stylelint-scss/stylelint-scss/issues> instead?
Thanks.

**Comment by jeddy3** [comment]
Fixed in stylelint-scss.
